### ISSUE TITLE
Attach tests don't run on mac osx

### ISSUE THREAD

**Issue by DanHeidinga** [eclipse/openj9#3206]
Test targets that fail:
AttachAPISanity_0

the problem is with `java.lang.ProcessBuilder`:
```
Caused by: java.lang.Error: Darwin is not a supported OS platform.
	at java.base/java.lang.ProcessImpl$Platform.get(ProcessImpl.java:165)
	at java.base/java.lang.ProcessImpl.<clinit>(ProcessImpl.java:169)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at java.base/java.lang.Runtime.exec(Runtime.java:635)
	at java.base/java.lang.Runtime.exec(Runtime.java:494)
	at org.openj9.test.attachAPI.TargetManager.launchTarget(TargetManager.java:248)
```
Note that the attach API tests by necessity launch child processes. This issue will probably also kill the VmArgument tests.

_Originally posted by @pdbain-ibm in https://github.com/eclipse/openj9/issues/3154#issuecomment-428566457_

**Comment by DanHeidinga** [comment]
@pdbain-ibm Can you dig into this further?  Is this a limitation in the extension code or in OpenJ9?

**Comment by pdbain-ibm** [comment]
@DanHeidinga will do.  I think this is a broader issue than attach API, so the title is misleading.

**Comment by stevewallin** [comment]
A number of the test failures for Mac are due to os.name being set to "Darwin" throughout the J9 implementation and "Mac OS X" in hotspot.  

OSInfo.java does explicit testing for "OS X" in the osName System Property and fails a number of the test due to returning an unknown platform from os.name

**Comment by DanHeidinga** [comment]
Thanks @stevewallin!  We'll look at getting that fixed asap

**Comment by DanHeidinga** [comment]
Change needs to be made to `omrsysinfo_get_OS_type ` in https://github.com/eclipse/omr/blob/2897f28e7ef6184f557c93798b5cdc6b720d59cd/port/unix/omrsysinfo.c

**Comment by pdbain-ibm** [comment]
The issue is `omrsysinfo_get_OS_type()` calls `uname()`, which returns "Darwin".
`vmprops.c:initializeSystemProperties()` calls `j9sysinfo_get_OS_type()` (which is an alias for the OMR function of the same name) and sets the "os.name" system property.

**Comment by DanHeidinga** [comment]
Discussed with Charlie.  Let's change it in OMR.  We'll need to validate this doesn't break any of the OMR tests as well

**Comment by pdbain-ibm** [comment]
Okay. I am testing a patch now and will run a personal build.

**Comment by DanHeidinga** [comment]
@pdbain-ibm Can you get the PR for this opened as well?  We can try and review in parallel then

**Comment by pdbain-ibm** [comment]
Your wish is my command: https://github.com/eclipse/omr/pull/3052

**Comment by stevewallin** [comment]
J9 also uses os.arch: amd64 and HS as os.arch: x86_64 - if users are testing these then there might be app/framework failures

**Comment by pdbain-ibm** [comment]
New pull request per comments on the old one: https://github.com/eclipse/omr/pull/3053

**Comment by pdbain** [comment]
It's working now:
```
PASSED: testAttachToOtherProcess
PASSED: testAttachToSelf
PASSED: testAttachToSelfAndOthers

openjdk version "11-internal" 2018-09-25
OpenJDK Runtime Environment (build 11-internal+0-adhoc.pdbain.openj9-openjdk-jdk11)
Eclipse OpenJ9 VM (build master-fa48180f, JRE 11 Mac OS X amd64-64-Bit Compressed References 20181013_000000 (JIT enabled, AOT enabled)
OpenJ9   - fa48180f
OMR      - 6f35a3ce
JCL      - 4f9e2ba5c4 based on jdk-11+28)
```
`org.openj9.test.attachAPI.TestAttachAPI.test_agntld01` is failing, but everything else seems to work.

**Comment by pshipton** [comment]
@pdbain-ibm is there another issue/pr for the failure? Should we close this one?

**Comment by pdbain-ibm** [comment]
https://github.com/eclipse/openj9/issues/3266 is tracking the agentld problem. Please close this issue.
